src/Backend/Synapse/DestinationTableOptions.php

```diff
@@ -9,29 +9,56 @@
  */
 final class DestinationTableOptions
 {
+    public const TABLE_DISTRIBUTION_HASH = 'HASH';
+    public const TABLE_DISTRIBUTION_ROUND_ROBIN = 'ROUND_ROBIN';
+
     /** @var String[] */
     private $columnNamesInOrder;
 
     /** @var String[] */
     private $primaryKeys;
 
+    /** @var string */
+    private $distribution;
+
+    /** @var String[] */
+    private $distributionColumnsNames;
+
     /**
      * @param String[] $columnNamesInOrder
      * @param String[] $primaryKeys
+     * @param String[] $distributionColumnsNames
      */
     public function __construct(
         array $columnNamesInOrder,
-        array $primaryKeys
+        array $primaryKeys,
+        string $distribution = self::TABLE_DISTRIBUTION_ROUND_ROBIN,
+        array $distributionColumnsNames = []
```

Comment:
Pohrávám si s myšlenkou jestli by pro ty dva typy distribuce neměly být vlastní `DestinationTableOptions`. Ale možná je to blbost.

Reply:
Jakože bys měl objekt `TableDistribution{$distributionName, $distributionKey}`

Reply:
Jako ne, ale vlastne i tohle je varianta. Jako ze `TableDistributionRoundRobin()` `TableDistributionHash($distributionColumnsNames)`?

Nebo `DestinationHashTableOptions` a `DestinationRoundRobinTableOptions` ?

Moc se mi nepozdava to pridavani dalsich parametru. To je cesta zpatky do 👿

Reply:
myslel sem spíš druhů variantu.

Já sem přemýšlal, že by nebyl špatný nějaký Table objekt v utils.
Něco jako
```
TableInterace{

ctor($columns,...)
pubsf fromReflection(TableRef):self

pubf getColumnsNames():arr
....
}
```
Pak by dávalo smysl objektově popsat aj with options.

Reply:
Upraveno https://github.com/keboola/php-db-import-export/pull/47/commits/3aa4cd3eb8b20b8075a32adc13dbdcacd3283024